food-truck-frontend/src/api/RateReviewApi.ts

```
@@ -95,59 +86,46 @@ export const loadReviewFromTruckForUser = async (
   try {
     // Get the reviews for this truck
     let resp: any = await api.request({
       url: `/reviews/truck/${truckId}`,
       method: "GET",
     });
 
-    // If we got a good response, then create frontend representation
-    if (resp.data !== null) {
-      // Map all backend reviews to frontend reviews
-      // Learned to use `Promise.all` from: https://stackoverflow.com/questions/40140149/use-async-await-with-array-map
-      reviews.reviews = await Promise.all(resp.data.map(async (r: any) => {
-        // Fetch username
-        let name: any = await api.request({
-          url: `/user/${r.userId}`,
-          method: "GET",
-        });
-        
-        // Create frontend representation for 1 review, with either the username or a blank string
-        return review_backendToFrontend(
-          r,
-          name.data !== null ? name.data.username : ""
-        );
-      }));
-    }
+    if (resp.data == null)
+      throw 'Truck not found';
+
+    // Map all backend reviews to frontend reviews
+    // Learned to use `Promise.all` from: https://stackoverflow.com/questions/40140149/use-async-await-with-array-map
+    const promises: Promise<Review | null>[] = resp.data.map(async (r: any) => {
+      const name = await getUsername(r.userId)
+      if (!name)
+        throw 'User not found';
+      const rev: Review | null = parse(ReviewMeta, review_backendToFrontend(r, name))
+      return rev;
+    })
+
+    const reviews: Review[] = (await Promise.all(promises))
+      .filter(r => r != null) as Review[];
 
-    // Fetch the truck's info. Yes, I could've gotten it from the reviews. That seemed like a coupling hazard, what happens
-    // if we change the way reviews get stored and they suddenly don't store trucks directly?
-    let truck: Truck | null = truck_backendToFrontend((await api.request({
-      url: `/truck/${truckId}`,
-      method: "GET"
-    })).data);
-    console.log(truck);
+    const truck = await getTruckById(truckId, () => {
```

Is there a reason we're swallowing the error here?